### PR TITLE
upgrade azblob sdk

### LINKED ISSUE #1947: Failed to sync large files into Azure Blob Storage, ErrorCode=RequestBodyTooLarge
**What happened**:

Using `juicefs sync` to copy files from AWS S3 to Azure Blob Storage, got error when the files are too large (not sure of the exact threshold). The smaller files work well.

From what I saw, files larger than 3GB failed, while ~200MB succeeded.

Error message:

```
Time:2022-05-02T17:55:58.7465947Z, RequestInitiated=Mon, 02 May 2022 17:55:57 GMT, RequestId=809d1e23-201e-0005-5f4d-5e36b1000000, API Version=2017-04-17, QueryParameterName=, QueryParameterValue=
2022/05/03 01:56:04.853836 juicefs[404967] <ERROR>: Failed to copy 0002_part_00.parquet: storage: service returned error: StatusCode=413, ErrorCode=RequestBodyTooLarge, ErrorMessage=The request body is too large and exceeds the maximum permissible limit.
```

### PR DESCRIPTION
**Pull request by zhijian-pro**
Close #1947

### CODE REVIEW

File: pkg/object/azure.go [hunk 1]
```diff
@@ -21,71 +21,69 @@ package object
...
 }
 
 func (b *wasb) String() string {
-	return fmt.Sprintf("wasb://%s/", b.container.Name)
+	return fmt.Sprintf("wasb://%s/", b.cName)
 }
 
 func (b *wasb) Create() error {
-	_, err := b.container.CreateIfNotExists(&storage.CreateContainerOptions{})
+	_, err := b.container.Create(ctx, nil)
```
Comment (davies) [inline pkg/object/azure.go:44]: Will it fail if it already exists?

File: pkg/object/azure.go [hunk 2]
```diff
@@ -96,51 +94,45 @@ func (b *wasb) List(prefix, marker string, limit int64) ([]Object, error) {
...
-func autoWasbEndpoint(containerName, accountName, accountKey string, useHTTPS bool) (string, error) {
-	baseURLs := []string{"core.windows.net", "core.chinacloudapi.cn"}
+func autoWasbEndpoint(containerName, accountName, scheme string, credential *azblob.SharedKeyCredential) (string, error) {
+	baseURLs := []string{"blob.core.windows.net", "blob.core.chinacloudapi.cn"}
 	endpoint := ""
 	for _, baseURL := range baseURLs {
-		client, err := storage.NewClient(accountName, accountKey, baseURL, "2017-04-17", useHTTPS)
+		client, err := azblob.NewContainerClientWithSharedKey(fmt.Sprintf("%s://%s.%s/%s", scheme, accountName, baseURL, containerName), credential, nil)
 		if err != nil {
 			log.Fatalf("Failed to create client: %v", err)
```
Comment (davies) [inline pkg/object/azure.go:128]: return err

File: pkg/object/azure.go (hunk 1 above)
Comment (zhijian-pro) [reply pkg/object/azure.go:44]: yes, already fixed

File: pkg/object/azure.go [hunk 3]
```diff
@@ -21,71 +21,73 @@ package object
...
 
 func (b *wasb) String() string {
-	return fmt.Sprintf("wasb://%s/", b.container.Name)
+	return fmt.Sprintf("wasb://%s/", b.cName)
 }
 
 func (b *wasb) Create() error {
-	_, err := b.container.CreateIfNotExists(&storage.CreateContainerOptions{})
+	_, err := b.container.Create(ctx, nil)
+	if strings.Contains(err.Error(), string(azblob.StorageErrorCodeContainerAlreadyExists)) {
```
Comment (davies) [inline pkg/object/azure.go:46]: Please add a test case to cover this

File: pkg/object/azure.go [hunk 4]
```diff
@@ -21,71 +21,73 @@ package object
...
-	return b.container.GetBlobReference(dst).Copy(uri, nil)
+	_, err := b.container.NewBlockBlobClient(dst).CopyFromURL(ctx, b.container.NewBlockBlobClient(src).URL(),
+		&azblob.CopyBlockBlobFromURLOptions{})
+	return err
 }
 
 func (b *wasb) Delete(key string) error {
-	return b.container.GetBlobReference(key).Delete(nil)
+	_, err := b.container.NewBlockBlobClient(key).Delete(ctx, &azblob.DeleteBlobOptions{})
+	if err != nil && strings.Contains(err.Error(), string(azblob.StorageErrorCodeBlobNotFound)) {
```
Comment (davies) [inline pkg/object/azure.go:87]: Can we use errors.Is(err, azblob.StorageErrorCodeBlobNotFound)

Comment (zhijian-pro) [reply pkg/object/azure.go:87]: No, `azblob.StorageErrorCodeBlobNotFound` is string type, does not implement error interface.

**Review by davies — APPROVED** [review]

**Pull request merged**